org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java

```diff
@@ -130,6 +130,9 @@ private void updateEditorAnnotations(@NonNull ISourceViewer sourceViewer, Publis
 	}
 
 	private void updateMarkers(PublishDiagnosticsParams diagnostics, IResource resource) throws CoreException {
+		if (!resource.isAccessible()) {
+			return;
```

Comment:
if we want to just skip silently this resource, can we not just remove the `LanguageServerPlugin.logError(e)` on line 176? That would be better for the performance, as testing for existence is not for free, and we expect to usually have the resource available. If need be, we can also check accessibility in addition to existence when catching the `CoreException`.

Reply:
Did you see some issue with current logging? I'm not against changing it, but if there is no actual reason to do it, I'd rather keep it separated from this change.

Reply:
No, I do not have a problem with the current logging. I have a problem only with calling  `!resource.isAccessible()` as a preemptive measure, since it has a cost and the condition will happen very rarely. I would prefer to be optimistic, and catch an exception if the resource is not accessible. To do that we would just adding `!resource.isAccessible()` to the condition on line 175 (or 172 before this change). It would be equivalent to this PR but creating a log entry, that is why I proposed to remove the log entry we created in that branch. If we are happy with the log entry (which I am), I would just move line 133 to line 175.

Note that line 175 already checks `resource.exists()`, which in some implementation is the same as `resource.isAccessible()`,  so I wonder if you have some stack trace available, maybe the existing catch is not on the right place?

Reply:
You're entirely right. I did change the patch to replace existing  `resource.exists()` by `resource.isAccessible()` and it provides a good result.